dss-collection-nft/contracts/DSSCollection.cdc

```diff
@@ -0,0 +1,526 @@
+        }
+
+        // getIDs returns an array of the IDs that are in the collection
+        //
+        pub fun getIDs(): [UInt64] {
+            return self.ownedNFTs.keys
+        }
+
+        // borrowNFT gets a reference to an NFT in the collection
+        //
+        pub fun borrowNFT(id: UInt64): &NonFungibleToken.NFT {
+            pre {
+                self.ownedNFTs[id] != nil: "Cannot borrow NFT, no such id"
+            }
+
+            return (&self.ownedNFTs[id] as &NonFungibleToken.NFT?)!
+        }
+
+        // borrowDSSCollectionNFT gets a reference to an NFT in the collection
+        //
+        pub fun borrowDSSCollectionNFT(id: UInt64): &DSSCollection.NFT? {
+            if self.ownedNFTs[id] != nil {
+                if let ref = &self.ownedNFTs[id] as auth &NonFungibleToken.NFT? {
+                    return ref! as! &DSSCollection.NFT
+                }
+                return nil
+            } else {
+                return nil
+            }
+        }
+
+        pub fun borrowViewResolver(id: UInt64): &AnyResource{MetadataViews.Resolver} {
+            let nft = (&self.ownedNFTs[id] as auth &NonFungibleToken.NFT?)!
+            let dssNFT = nft as! &DSSCollection.NFT
+            return dssNFT as &AnyResource{MetadataViews.Resolver}
+        }
+
+        destroy() {
+            destroy self.ownedNFTs
+        }
+
+        init() {
+            self.ownedNFTs <- {}
+        }
+    }
+
+    // public function that anyone can call to create a new empty collection
+    //
+    pub fun createEmptyCollection(): @NonFungibleToken.Collection {
+        return <- create Collection()
+    }
+
+    // An interface containing the Admin function that allows minting NFTs
+    //
+    pub resource interface NFTMinter {
+        // Mint a single NFT
+        // The collectionGroupID for the given ID must already exist
+        //
+        pub fun mintNFT(collectionGroupID: UInt64, completedBy: String, level: UInt64): @DSSCollection.NFT
+    }
+
+    // A resource that allows managing metadata and minting NFTs
+    //
+    pub resource Admin: NFTMinter {
+
+        // Borrow a Collection Group
+        //
+        pub fun borrowCollectionGroup(id: UInt64): &DSSCollection.CollectionGroup {
+            pre {
+                DSSCollection.collectionGroupByID[id] != nil: "Cannot borrow collection group, no such id"
+            }
+
+            return (&DSSCollection.collectionGroupByID[id] as &DSSCollection.CollectionGroup?)!
+        }
+
+        // Create a Collection Group
+        //
+        pub fun createCollectionGroup(
+            name: String,
+            productPublicPath: PublicPath,
+            startTime: UFix64?,
+            endTime: UFix64?,
+            timeBound: Bool
+        ): UInt64 {
+            // Create and store the new collection group
+            let collectionGroup <- create DSSCollection.CollectionGroup(
+                name: name,
+                productPublicPath: productPublicPath,
```

Comment:
what's the `productPublicPath` used for?

Reply:
I have changed this to typeName which is the contract. Both were there to signal the product / owner of the Collection Group. I used product path when I intended to do on-chain validation, but Dete preferred this on the BE impl. 

Changed to typeName on suggestion from Ryan T.